src/gan/gen_pyt/common/utils.py

```diff
@@ -0,0 +1,148 @@
+# coding=utf-8
+import argparse
+
+
+class cached_property(object):
+    """ A property that is only computed once per instance and then replaces
+        itself with an ordinary attribute. Deleting the attribute resets the
+        property.
+
+        Source: https://github.com/bottlepy/bottle/commit/fa7733e075da0d790d809aa3d2f53071897e6f76
+        """
+
+    def __init__(self, func):
+        self.__doc__ = getattr(func, '__doc__')
+        self.func = func
+
+    def __get__(self, obj, cls):
+        if obj is None:
+            return self
+        value = obj.__dict__[self.func.__name__] = self.func(obj)
+        return value
+
+
+def init_arg_parser():
```

Comment:
I assume all of this was coded by someone else. if not, it might make more sense to turn this to a config.

Reply:
Actually in process implementing that as we speak! Diana had a question about the file that calls this which highlighted the issue for me, too.